src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1429,6 +1435,23 @@ public char FirstCharOfOneOrMulti()
             }
         }
 
+        /// <summary>Data about a starting literal as returned by <see cref="FindStartingLiteral"/>.</summary>
+        public readonly struct StartingLiteralData
```

Comment:
If we want to define in one line, we could have this be a `readonly record struct` right?

Reply:
We could.  That just brings with it a bunch of other stuff that's not actually needed here, so while very convenient and great for higher level code, in general I've stayed away from using records in our core libraries.